[BUG]: dist-types/index.d.ts is not present in the published NPM package

### What happened?

Hey awesome maintainers! After upgrading from 4.1.1 to 4.1.2 I noticed the following tsc error:
```shell
 error TS7016: Could not find a declaration file for module '@octokit/plugin-retry'. '/some-path-here/node_modules/.pnpm/@octokit+plugin-retry@4.1.2_@octokit+core@4.2.0/node_modules/@octokit/plugin-retry/dist-node/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/octokit__plugin-retry` if it exists or add a new declaration (.d.ts) file containing `declare module '@octokit/plugin-retry';`

2 import { retry } from '@octokit/plugin-retry';
```
Having a look in the tgz on npm I only see a `dist-node` directory, but the package.json exports the types in `dist-types/index.d.ts`


### Versions

"@octokit/core": "4.2.0"
"@octokit/plugin-retry": "4.1.2",
"@octokit/rest": "19.0.7",
node 16.19.1

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -15,6 +15,8 @@ jobs:
           cache: npm
       - run: npm ci
       - run: npm run build
+      - name: "Fix pkg.files file pattern"
+        run: node scripts/fix-package-json.js
       - run: npx semantic-release
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

--- a/scripts/fix-package-json.js
+++ b/scripts/fix-package-json.js
@@ -1,0 +1,15 @@
+const fs = require("fs");
+const path = require("path");
+const {EOL} = require("os");
+
+const pkgPath = path.join(__dirname, "../pkg/package.json");
+const pkg = JSON.parse(fs.readFileSync(pkgPath, "utf8"));
+
+pkg.files = pkg.files.map((file) => {
+  if (file.endsWith("/")) {
+    return file + "**";
+  }
+  return file;
+});
+
+fs.writeFileSync(pkgPath, JSON.stringify(pkg, null, 2) + EOL, "utf8");

